test/functional/Valhalla/src/org/openj9/test/lworld/ValueTypeGenerator.java

```diff
@@ -198,158 +192,147 @@ private static void testWithFieldOnNonExistentClass(ClassWriter cw, String class
 		mv.visitEnd();
 	}
 	
-	private static void makeValueHelper(ClassWriter cw, String valueName, String makeValueSig, String makeValueGenericSig, String[] fields, int makeMaxLocal) {
-		makeGeneric(cw, valueName, "makeValueGeneric", "makeValue", makeValueSig, makeValueGenericSig, fields, makeMaxLocal);
+	private static void makeValue(ClassWriter cw, String valueName, String makeValueSig, String[] fields, int makeMaxLocal) {
 		
-		boolean doubleDetected = false;
 		MethodVisitor mv = cw.visitMethod(ACC_PUBLIC  + ACC_STATIC, "makeValue", "(" + makeValueSig + ")L" + valueName + ";", null, null);
 		mv.visitCode();
 		mv.visitTypeInsn(DEFAULTVALUE, valueName);
-		for (int i = 0; i <  fields.length; i++) {
+		for (int i = 0, count = 0; i <  fields.length; i++) {
 			String nameAndSig[] = fields[i].split(":");
 			switch (nameAndSig[1]) {
 			case "D":
-				doubleDetected = true;
-				mv.visitVarInsn(DLOAD, i);
+				mv.visitVarInsn(DLOAD, count);
+				count += 2;
 				break;
 			case "I":
 			case "Z":
 			case "B":
 			case "C":
 			case "S":
-				mv.visitVarInsn(ILOAD, i);
+				mv.visitVarInsn(ILOAD, count);
+				count++;
 				break;
 			case "F":
-				mv.visitVarInsn(FLOAD, i);
+				mv.visitVarInsn(FLOAD, count);
+				count++;
 				break;
 			case "J":
-				doubleDetected = true;
-				mv.visitVarInsn(LLOAD, i);
+				mv.visitVarInsn(LLOAD, count);
+				count += 2;
 				break;
 			default:
-				mv.visitVarInsn(ALOAD, i);
+				mv.visitVarInsn(ALOAD, count);
+				count++;
 				break;
 			}
 			mv.visitFieldInsn(WITHFIELD, valueName, nameAndSig[0], nameAndSig[1]);
 		}
 		mv.visitInsn(ARETURN);
 		
-		int maxStackAndLocals = (doubleDetected) ? 3 : 2;
-		mv.visitMaxs(maxStackAndLocals, fields.length);
+		mv.visitMaxs(makeMaxLocal, makeMaxLocal);
```

Comment:
`mv.visitMaxs` takes in maxStack and maxLocal.

The maxStack if there is a double is 3 (2 for single) becuase everytime withfield is run it pops the arg of the stack